EntranceShuffle.py

```diff
@@ -624,7 +624,7 @@ def validate_world(world, worlds, entrance_placed, locations_to_ensure_reachable
                 if not max_search.visited(location):
                     raise EntranceShuffleError('%s is unreachable' % location.name)
 
-    if world.shuffle_interior_entrances and \
+    if world.shuffle_interior_entrances and (world.misc_hints or world.hints != 'none') and \
```

Comment:
Imo, this change is unrelated to the Misc Hints setting and should be kept for a separate PR. Currently, the entrance conditions here are unconditional to hint settings, so if we follow the current rules, it should remain that way regardless of the Misc. Hints setting.

In theory, we could indeed remove these special rules if there are no hints, but I think it requires discussion, and most people who would want to discuss it aren't going to expect it to come from this set of changes.